src/test/java/seedu/address/testutil/EditPersonDescriptorBuilder.java

```diff
@@ -76,7 +80,16 @@ public EditPersonDescriptorBuilder withAddress(String address) {
      * that we are building.
      */
     public EditPersonDescriptorBuilder withTags(String... tags) {
-        Set<Tag> tagSet = Stream.of(tags).map(Tag::new).collect(Collectors.toSet());
+        Set<Tag> tagSet = Stream.of(tags).map(tagName -> {
+                    if (PersonTagType.isValidPersonTag(tagName)) {
+                        return new PersonTag(tagName);
+                    } else if (PropertyTagType.isValidPropertyTag(tagName)) {
+                        return new PropertyTag(tagName);
+                    } else {
+                        return new Tag(tagName);
```

Comment:
Same here as above, consider maybe having a function under `Tag` or call it `TagUtil`.